### PR TITLE
1438 error when missing excel column

### PR DESCRIPTION
**Pull request by georgeDaskalakis**

### CODE REVIEW

File: src/OSPSuite.Core/Import/DataFormatParameter.cs [hunk 1]
```diff
@@ -171,7 +171,7 @@ public override bool IsGroupingCriterion()
 
       public override bool ComesFromColumn()
       {
-         return false;
+         return true;
```
Comment (msevestre) [inline src/OSPSuite.Core/Import/DataFormatParameter.cs:174]: Lol. Drastic change. :)

File: src/OSPSuite.Infrastructure.Import/Core/DataFormat/AbstractColumnsDataFormat.cs [hunk 2]
```diff
@@ -209,6 +209,10 @@ protected virtual void ExtractGeneralParameters(List<string> keys, IUnformattedD
       public IEnumerable<ParsedDataSet> Parse(IUnformattedData data,
          IReadOnlyList<ColumnInfo> columnInfos)
       {
+         var missingColumns = Parameters.Where(p => p.ComesFromColumn() && data.GetColumnDescription(p.ColumnName) == null).Select(p => p.ColumnName);
+         if (missingColumns.Any())
```
Comment (msevestre) [inline src/OSPSuite.Infrastructure.Import/Core/DataFormat/AbstractColumnsDataFormat.cs:213]: You are iterating here already . So call the toList before on line 212 and not on 214.

File: src/OSPSuite.Infrastructure.Import/Core/DataFormat/AbstractColumnsDataFormat.cs [hunk 3]
```diff
@@ -325,7 +326,7 @@ private SimulationPoint parseMappingOnSameColumn(MappingDataFormatParameter curr
             if (!string.IsNullOrEmpty(currentParameter.MappedColumn.Unit.ColumnName))
                unitColumnDescription = data.GetColumnDescription(currentParameter.MappedColumn.Unit.ColumnName);
             if (unitColumnDescription == null)
-               throw new MissingColumnException(currentParameter.MappedColumn.Unit.ColumnName);
+               throw new MissingColumnException(new List<string>() { currentParameter.MappedColumn.Unit.ColumnName });
```
Comment (msevestre) [inline src/OSPSuite.Infrastructure.Import/Core/DataFormat/AbstractColumnsDataFormat.cs:329]: Snce you own this exception, create an overload

**Review by msevestre — CHANGES_REQUESTED** [review]
A few changes but nothing hard

File: src/OSPSuite.Core/Import/DataFormatParameter.cs (hunk 1 above)
Comment (georgeDaskalakis) [reply src/OSPSuite.Core/Import/DataFormatParameter.cs:174]: yeap, was wrong before, although it was not causing any errors

File: src/OSPSuite.Infrastructure.Import/Core/DataFormat/AbstractColumnsDataFormat.cs (hunk 2 above)
Comment (georgeDaskalakis) [reply src/OSPSuite.Infrastructure.Import/Core/DataFormat/AbstractColumnsDataFormat.cs:213]: done

File: src/OSPSuite.Infrastructure.Import/Core/DataFormat/AbstractColumnsDataFormat.cs (hunk 3 above)
Comment (georgeDaskalakis) [reply src/OSPSuite.Infrastructure.Import/Core/DataFormat/AbstractColumnsDataFormat.cs:329]: done

File: src/OSPSuite.Infrastructure.Import/Core/MissingColumnException.cs [hunk 4]
```diff
@@ -8,5 +8,9 @@ public class MissingColumnException : AbstractImporterException
       public MissingColumnException(IReadOnlyList<string> missingColumns) : base(Error.MissingColumnException(missingColumns))
       {
       }
+
+      public MissingColumnException(string missingColumn) : base(Error.MissingColumnException(new List<string>() { missingColumn}))
```
Comment (msevestre) [inline src/OSPSuite.Infrastructure.Import/Core/MissingColumnException.cs:12]: Ok
Better. Call this(new list...) That way you do not have to duplicate the logic.
Merging.

**Review by msevestre — APPROVED** [review]

**Pull request merged**